File: python/test/test_adjacent_difference.py
import ThrustRTC as trtc



darr1_in = trtc.device_vector_from_list([ 1, 2, 1, 2, 1, 2, 1, 2], 'int32_t')
darr1_out = trtc.device_vector('int32_t', 8)
trtc.Adjacent_Difference(darr1_in, darr1_out)
print (darr1_out.to_host())

darr2_in = trtc.device_vector_from_list([ 1, 2, 1, 2, 1, 2, 1, 2], 'int32_t')
darr2_out = trtc.device_vector('int32_t', 8)

trtc.Adjacent_Difference(darr2_in, darr2_out, trtc.Plus())
print (darr2_out.to_host())

File: java/test/test_adjacent_difference.java
import java.util.*;
import JThrustRTC.*;

public class test_adjacent_difference
{
	public static void main(String[] args) 
	{
        {
            DVVector vec_in = new DVVector(new int[] { 1, 2, 1, 2, 1, 2, 1, 2 });
            DVVector vec_out = new DVVector("int32_t", 8);
            TRTC.Adjacent_Difference(vec_in, vec_out);
            System.out.println(Arrays.toString((int[])vec_out.to_host()));
        }

        {
            DVVector vec_in = new DVVector(new int[] { 1, 2, 1, 2, 1, 2, 1, 2 });
            DVVector vec_out = new DVVector("int32_t", 8);
            TRTC.Adjacent_Difference(vec_in, vec_out, new Functor("Plus"));
            System.out.println(Arrays.toString((int[])vec_out.to_host()));
        }
	}	

}
